### PR TITLE
Transfer mypy settings to pyproject.toml file

### PR DESCRIPTION
**Pull request by sinderoth**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

I transferred the existing settings for mypy to the pyproject.toml file from mypy.ini file. 

Used `mypy .` command to test the changes .

### Related issue

- #513

### CODE REVIEW

**Comment by andrewtavis** [comment]
Thanks for this, @sinderoth! Appreciate the help here :)

Can you take a look, @to-sta?

**Comment by to-sta** [comment]
@sinderoth thanks for the PR. 

There are few minor issues with the new setup, but it's already nice just having one config file in the backend directory 😃.

File: backend/mypy.ini [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-mypy_path = backend
-plugins =
-    mypy_django_plugin.main,
-    mypy_drf_plugin.main
-ignore_missing_imports = True
-strict = True
-exclude = content/factory.py
-
-[mypy-content.*]
-follow_imports = skip
```
Comment (to-sta) [inline backend/mypy.ini:11]: `exclude` and `follow_imports` are missing in the `pyproject.toml`

File: backend/pyproject.toml [hunk 2]
```diff
@@ -58,3 +58,15 @@ omit = [
...
+plugins = [
+    mypy_django_plugin.main,
+    mypy_drf_plugin.main
+]
+ignore_missing_imports = True
+strict = True
+
+[[tool.mypy.overrides]]
+module = 'django-stubs'
+django_settings_module = 'backend.settings'
```
Comment (to-sta) [inline backend/pyproject.toml:72]: You need to use quotes and general we use double quotes inside the pyproject.toml. 

```toml
plugins = [
    "mypy_django_plugin.main",
    "mypy_drf_plugin.main"
]
```

`True` needs to be lower case inside pyproject.toml. Will indicate an error when you run mypy 

`pyproject.toml: Invalid value`

for  django-stubs have a look at the github repo, they have an example for the [pyproject.toml](https://github.com/typeddjango/django-stubs)

**Review by to-sta — CHANGES_REQUESTED** [review]

**Comment by sinderoth** [comment]
I added the missing parts but I don't think I can test this properly.  I first docker exec backend container and run "mypy ." but before and after the change this is what I get : "Success: no issues found in 6 source files"

**Comment by sinderoth** [comment]
thanks for the help, I think it should work now.

**Review by to-sta — APPROVED** [review]
Looks good @sinderoth 😃

**Review by andrewtavis — APPROVED** [review]
Thanks for the commit here, @sinderoth! And thanks for the review and helping along the way, @to-sta! 😊😊

**Pull request merged**